libraries/core_libs/network/src/tarcap/packets_handlers/transaction_packet_handler.cpp

```diff
@@ -76,12 +76,14 @@ inline void TransactionPacketHandler::process(const PacketData &packet_data, con
         }
         case TransactionStatus::InsufficentBalance:
         case TransactionStatus::LowNonce: {
-          if (peer->reportSuspiciousPacket()) {
+          // Raise exception in trx pool is over the limit and this peer already has too many suspicious packets
+          if (peer->reportSuspiciousPacket() && trx_mgr_->nonProposableTransactionsOverTheLimit()) {
```

Comment:
So we won't disconnect from peer that is sending a lot of suspicious packets(over peer limit) until we will reach this global limit. Why is this better then logic we had before?